allennlp/modules/token_embedders/embedding.py

```diff
@@ -140,6 +149,25 @@ def forward(self, inputs):  # pylint: disable=arguments-differ
             embedded = projection(embedded)
         return embedded
 
+    @overrides
```

Comment:
@matt-gardner,  I agree that full pretrained embedding based extension would be difficult for the reasons you mentioned in the issue. However, would it be worth exposing the API in `extend_vocab` something like :
`extend_vocab(extended_vocab, vocab_namespace = None, **pretrained_file = None**)`? This won't directly be usable via `fine-tune` command since we need to carry over pretrained-embedding-file. But if a user has access to pretrained-file, then with `extend_vocab` API, it can be easy to write a script to make a pretrained model compatible with pre-generated extended vocabulary. Solves my usecase at least : ) I am happy to keep it in separate pr as well.

Reply:
I think for ease of separating reviewing into logically-coherent pieces, it's better to just do that in another PR after this one is merged.  It also seems totally fine to me to at least make it possible to get this working in your own script first, then extend that functionality to having stuff work more generally later, so that PR can be relatively simple.

Reply:
And, I'll probably get to reviewing this tomorrow.

Reply:
no problem, thanks!